frontend/src/App.tsx

```diff
@@ -1,15 +1,33 @@
-import "@fontsource/hammersmith-one";
+import React from 'react'
+import { Outlet, useLocation } from 'react-router-dom';
 import "./App.scss";
-import HomePage from "./components/HomePage/HomePage";
+import "@fontsource/hammersmith-one";
 import Navbar from "./components/Navbar/Navbar";
+import LoadingScreen from './components/LoadingScreen/LoadingScreen';
+import PannellumBackground from './components/PannellumBackground/PannellumBackground';
 
 function App() {
+  const location = useLocation();
+
+  // List of routes that do not include the Navbar
+  const hideNavbarRoutes = ["/login", "/register"];
+  const panoBackgroundRoutes = ["/login", "/register", "/profile", "/home"]
+
+  // Determine if the Navbar should be hidden
+  const hasNavbar = !hideNavbarRoutes.includes(location.pathname);
+  const hasBackground = panoBackgroundRoutes.includes(location.pathname)
+
+  const [isPanoramaLoaded, setIsPanoramaLoaded] = React.useState(!hasBackground);
```

Comment:
i imagine vite tree shaking would make this change redundant but is it worth just importing exclusively `useState` from `React` instead of the whole `React` package? that would also be in-line with our components and stuff and how we do the same thing there